### DISCUSSION TITLE
Arch linux Install

### DISCUSSION THREAD

**Discussion by basilverthorn** [mathoudebine/turing-smart-screen-python#311] [General]
Anyone have instructions on how to install this in Arch. I have tried the python pip install on the wiki, but it fails instantly.  I am totaly a newb to linux and python, but trying to make the switch. The error I get is:

python3 -m pip install -r requirements.txt
error: externally-managed-environment

× This environment is externally managed
╰─> To install Python packages system-wide, try 'pacman -S
python-xyz', where xyz is the package you are trying to
install.

If you wish to install a non-Arch-packaged Python package,
create a virtual environment using 'python -m venv path/to/venv'.
Then use path/to/venv/bin/python and path/to/venv/bin/pip.

If you wish to install a non-Arch packaged Python application,
it may be easiest to use 'pipx install xyz', which will manage a
virtual environment for you. Make sure you have python-pipx
installed via pacman.

**Comment by mathoudebine** [comment]
Hi, on Arch Linux the Python packages are managed by Pacman the package manager, so you cannot install packages manually from Python Pip.
The solution is to create a virtual environment where you can install Python packages only for this project.
In the source directory of this project run:
```bash
python3 -m venv venv
```
You should now have a new `venv` folder that will contain your local Python virtual environment to run this project.
Everytime you want to run this project, you need first to activate this virtual environment from the terminal:
```bash
source venv/bin/activate
```
Then you can follow the rest of the documentation to install dependencies and run the project

PS: if you're not familiar with Linux you may find Arch Linux is not the best distribution to start with. It is very technical and a lot must be done by the user (like this for instance). I would recommend a more user-friendly distribution like Ubuntu to begin with.

**Reply by basilverthorn** [reply]
Thank you, that got me going. Thank you again.

**Comment by gregfr** [comment]
Thanks @mathoudebine for the explanation; note that it works also for other linux when `brew` is used.
